### ISSUE TITLE
[LINT_BUG]: unused object in data.table scope

### ISSUE THREAD

**Issue by Gotfrid** [Appsilon/box.linters#140]
### box.linters version

```
0.10.2
```

### Sample source code to lint

```R
threshold <- 0.5
data <- data.table::data.table(value = c(0.3, 0.6))
data[, color_level := ifelse(value <= threshold, 1, 0)]
```

### Lint command used

```
rhino::lint_r()
```

### Lint result

```
app/logic/utils.R:39:10: warning: [unused_declared_object_linter] Declared function/object unused.
  data[, color_level := ifelse(value <= threshold, 1, 0)]
         ^~~~~~~~~~~
```

### Expected result

No linter issues.

**Comment by shahreyar-abeer** [comment]
Hi,

I'd like to report a similar issue with the dot(.) operator in {data.table}

### box.linters version
```
0.10.0
```

### sample source code to lint
```
box::use(
  data.table[data.table, ],
)

d1 <- data.table(v = 1:9)
d1[, .(sum(v))]
```

### lint result
```
Line 6 [box_usage_linter] Function not imported nor defined.
```

### expected result
```
No linter issues
```

**Comment by radbasa** [comment]
This looks like custom special handlers for `{data.table}` are needed.

**Comment by radbasa** [comment]
@Gotfrid will need you help with this one. It would be great if you could write unit tests for data.table. I am not familiar with data.table and so am not confident in writing good tests.

**Comment by dereckmezquita** [comment]
@radbasa @Gotfrid, I wouldd like to help with this. I've been working on box.linters recently (PR #189 for the working dir and R6 inheritance fixes) and hit this exact issue in my own codebase.

I have a working implementation that handles:

1. **`:=` operator** -- parsed as `LEFT_ASSIGN` by R's parser. Fixed in `get_declared_objects()` by filtering `LEFT_ASSIGN[not(text()=':=')]`. This matches how lintr handles it in `implicit_assignment_linter.R`.
2. **Special symbols** (`.N`, `.SD`, `.I`, `.GRP`, `.BY`, `.NGRP`, `.EACHI`) -- added to an allowlist in `box_usage_linter` so they're not flagged as "Function not imported nor defined".
3. **`.()` alias for `list()`** -- included in the same allowlist.
4. **`..varname` prefix** -- excluded via `startsWith(text, "..")` check.
5. **`` `:=`() `` functional form** -- included in the allowlist.

This is a hardcoded approach (not configurable), which matches how lintr itself handles data.table in `object_usage_linter.R` and `shared_constants.R`. All 446 existing tests pass + 2 new tests added.

Would you prefer this as separate from #189 or combined?

- https://github.com/dereckmezquita/box.linters/tree/fix/data-table-support
